FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java

```diff
@@ -68,6 +80,147 @@ public void setUser(UserObservable user) {
         this.user = user;
     }
 
+
+    private void populate_friends(){
+        ParseQuery<Friends> query = ParseQuery.getQuery(Friends.class);
+        query.include(Friends.REQUESTED_KEY);
+        query.whereEqualTo(Friends.USER_KEY, user.getUser());
+        List<Friends> parse_friends = new ArrayList<Friends>();
+        try {
+            parse_friends = query.find();
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        for (Friends friend: parse_friends){
+            friends.add(friend.getUser());
+        }
+    }
+
+    /**
+     * Adds restaurant from parse or calls yelpquery to do so depending on parse_source
+     */
+    public void findRestaurants(String apiKey, @Nullable RestaurantListener listener) {
+        if (source == PARSE_LIKES){
+            parseLikes(apiKey, listener);
+        }
+        else if (source == PARSE_TOGOS){
+            parseTogos(apiKey, listener);
+        }
+        else if (source == PARSE_GENERAL){
+            parseGeneral(apiKey, listener);
+        } else {
+            yelpQuery(apiKey, listener);
+        }
+    }
+
+    private void parseLikes(String apiKey, RestaurantListener listener) {
+        ParseQuery<UserLike> query = ParseQuery.getQuery(UserLike.class);
```

Comment:
Same with a LikesHelper

Reply:
Don't think I'll do this just cause don't want to overdo it, as function is returning restaurants, which makes sense to do in restaurant server